File: Python/Easy/1460.py
class Solution:
    def canBeEqual(self, target: List[int], arr: List[int]) -> bool:
        # use a dictionary to store the frequency of one array then compare with the other array
        freq = {}

        # create the frequency dictionary
        for i in target:
            temp = freq.get(i, 0)
            temp += 1
            freq[i] = temp

        # compare with the other array
        for i in arr:
            temp = freq.get(i, 0)
            if temp == 0:
                return False
            temp -= 1
            freq[i] = temp

        return True

File: Java/Easy/1460.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        // use a hashmap to store frequency of one array then compare with the other
        Map<Integer, Integer> map = new HashMap<>();
        int temp;

        // create the frequency count
        for (int num : target) {
            temp = map.getOrDefault(num, 0);
            temp++;
            map.put(num, temp);
        }

        // compare
        for (int num : arr) {
            temp = map.getOrDefault(num, 0);
            if (temp == 0) {
                return false;
            }
            temp--;
            map.put(num, temp);
        }
        return true;
    }
}